### PR TITLE
improve the GitHub Server subdomain isolation config description

### LINKED ISSUE #22699: Improve the documentation about configuring GitHub server with subdomain isolation
### Is your enhancement related to a problem? Please describe

The current documentation about the subdomain isolation annotation (https://eclipse.dev/che/docs/stable/administration-guide/configuring-oauth-2-for-github/#applying-the-github-oauth-app-secret) is confusing. It is not obvious what value to setup for this annotation.

### Describe the solution you'd like

* Add some description about the subdomain isolation option of the GitHub Enterprise Server.
* Explain how to setup the annotations with specific cases.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by vinokurig**
## What does this pull request change?
improve the GitHub Server subdomain isolation config description:
screenshot-0 0 0 0_4000-2023 11 29-15_43_48

## What issues does this pull request fix or reference?
fixes https://github.com/eclipse/che/issues/22699

## Specify the version of the product this pull request applies to
3.11
## Pull Request checklist

The author and the reviewers validate the content of this pull request with the following checklist, in addition to the [automated tests](code_review_checklist.adoc).

- Any procedure:
  - [ ] Successfully tested.
- Any page or link rename:
  - [ ] The page contains a redirection for the previous URL.
  - Propagate the URL change in:
    - [ ] Dashboard [default branding data](https://github.com/eclipse-che/che-dashboard/blob/main/packages/dashboard-frontend/src/services/bootstrap/branding.constant.ts)
- [ ] Builds on [Eclipse Che hosted by Red Hat](https://workspaces.openshift.com).
- [ ] the *`Validate language on files added or modified`* step reports no vale warnings.

### CODE REVIEW

File: modules/administration-guide/partials/proc_applying-the-github-oauth-app-secret.adoc [hunk 1]
```diff
@@ -36,15 +36,15 @@ metadata:
...
 type: Opaque
 stringData:
   id: __<GitHub_OAuth_Client_ID>__ <4>
   secret: __<GitHub_OAuth_Client_Secret>__ <5>
 ----
 <1> The {prod-short} namespace. The default is `{prod-namespace}`.
 <2> This depends on the GitHub product your organization is using: When hosting repositories on GitHub.com or GitHub Enterprise Cloud, omit this line or enter the default `https://github.com`. When hosting repositories on GitHub Enterprise Server, enter the GitHub Enterprise Server URL.
-<3> This line is only added for GitHub Enterprise Server. To disable subdomain isolation, set this to `"true"`. To enable subdomain isolation, set this to `"false"`.
+<3> This line is only added for GitHub Enterprise Server with disabled https://docs.github.com/en/enterprise-server/admin/configuration/hardening-security-for-your-enterprise/enabling-subdomain-isolation#about-subdomain-isolation[subdomain isolation] option. If the https://docs.github.com/en/enterprise-server/admin/configuration/hardening-security-for-your-enterprise/enabling-subdomain-isolation#about-subdomain-isolation[subdomain isolation] option is enabled on GitHub Enterprise Server, this annotation must be either omitted or set to `false`.
```
Comment (deerskindoll) [inline modules/administration-guide/partials/proc_applying-the-github-oauth-app-secret.adoc]:
```suggestion
<3> This line is only added for GitHub Enterprise Server with disabled link:https://docs.github.com/en/enterprise-server/admin/configuration/hardening-security-for-your-enterprise/enabling-subdomain-isolation#about-subdomain-isolation[subdomain isolation] option. If the subdomain isolation option is enabled on GitHub Enterprise Server, you must either omit this annotation or set it to `false`.
```

**Review by deerskindoll — COMMENTED** [review]
fixed link and a small edit

**Review by ibuziuk — APPROVED** [review]

**Comment by vinokurig** [comment]
@deerskindoll I have fixed your suggestion, could you please review the PR again?

**Review by deerskindoll — APPROVED** [review]
lgtm

**Pull request merged**